### PR TITLE
Add docstring to functions in hawk_test

### PR DESCRIPTION
**Pull request by Dawei-Pang**
According to the request in https://jira.suse.com/browse/TEAM-9131, add doc strings for all functions in all python scripts within the repository to summary what the functions do.

### CODE REVIEW

File: e2e_test/hawk_test_driver.py [hunk 1]
```diff
@@ -92,7 +92,20 @@ class LongLiterals:
 
 
 class HawkTestDriver:
+    '''
+    Hawk Test driver main class
+    '''
     def __init__(self, addr='localhost', port='7630', browser='firefox', headless=False, version='15-SP5'):
+        '''
+        Constructor function to initlize parameters
```
Comment (alvarocarvajald) [inline e2e_test/hawk_test_driver.py:100]: s/initlize/initialize/

File: e2e_test/hawk_test_driver.py [hunk 2]
```diff
@@ -106,6 +119,12 @@ def __init__(self, addr='localhost', port='7630', browser='firefox', headless=Fa
             self.timeout_scale = 1
 
     def _connect(self):
+        '''
+        Private method: Initlize webdriver basing on browser type
```
Comment (alvarocarvajald) [inline e2e_test/hawk_test_driver.py:123]: s/Initlize/Initialize/

File: e2e_test/hawk_test_driver.py [hunk 3]
```diff
@@ -181,6 +245,11 @@ def find_element(self, bywhat, texto, tout=60):
         return elem
 
     def verify_success(self):
+        '''
+        Function to find and verify success
```
Comment (alvarocarvajald) [inline e2e_test/hawk_test_driver.py:249]:
```suggestion
        Function to find alert-success element on the UI.
```

File: e2e_test/hawk_test_driver.py [hunk 4]
```diff
@@ -189,6 +258,14 @@ def verify_success(self):
         return True
 
     def fill_value(self, field, tout):
+        '''
+        Function to find and fill value
```
Comment (alvarocarvajald) [inline e2e_test/hawk_test_driver.py:262]:
```suggestion
        Function to find an input element in the UI by its field name, and fill in a value.
```

File: e2e_test/hawk_test_driver.py [hunk 5]
```diff
@@ -189,6 +258,14 @@ def verify_success(self):
...
+        Args:
+            field (str) : target field
+            tout  (str) : send key
```
Comment (alvarocarvajald) [inline e2e_test/hawk_test_driver.py:265]:
```suggestion
            tout  (str) : text to input
```

File: e2e_test/hawk_test_driver.py [hunk 6]
```diff
@@ -197,19 +274,37 @@ def fill_value(self, field, tout):
...
-    # Internal support function to find element(s) by xpath and click them
-    # Sets test_status to False on failure.
     def check_and_click_by_xpath(self, errmsg, xpath_exps):
+        '''
+        Internal support function to find element(s) by xpath and click them
+        Sets test_status to False on failure.
+
+        Args:
+           errmsg (str): error message
+           xpath_exps (str): xpath express
```
Comment (alvarocarvajald) [inline e2e_test/hawk_test_driver.py:303]: s/express/expression/

File: e2e_test/hawk_test_driver.py [hunk 7]
```diff
@@ -226,6 +321,14 @@ def check_and_click_by_xpath(self, errmsg, xpath_exps):
 
     # Generic function to perform the tests
     def test(self, testname, results, *extra):
+        '''
+        Generic function to perform the tests
```
Comment (alvarocarvajald) [inline e2e_test/hawk_test_driver.py:325]: Suggestion:
```
Main test caller. This is essentially a wrapper to call each specific test.
Each test performs different actions, but this function will wrap around
them to perform common actions which are required by all tests such
as connecting to the webUI, authenticating, setting the test result and
disconnecting from the UI.
```

File: e2e_test/hawk_test_driver.py [hunk 8]
```diff
@@ -226,6 +321,14 @@ def check_and_click_by_xpath(self, errmsg, xpath_exps):
...
+
+        Args:
+            testname (str): test name
+            results :       result object
+            *extra  :       extra parameters
```
Comment (alvarocarvajald) [inline e2e_test/hawk_test_driver.py:330]:
```suggestion
            *extra  :       extra parameters required by some of the specific tests which this function calls.
```

File: e2e_test/hawk_test_driver.py [hunk 9]
```diff
@@ -255,6 +363,12 @@ def test_set_stonith_maintenance(self):
         return False
 
     def test_disable_stonith_maintenance(self):
+        '''
+        Set disable STONITH/sbd in maintenance
```
Comment (alvarocarvajald) [inline e2e_test/hawk_test_driver.py:367]:
```suggestion
        Disable maintenance in STONITH/sbd
```

File: e2e_test/hawk_test_driver.py [hunk 10]
```diff
@@ -312,6 +450,15 @@ def test_disable_maintenance_first_node(self):
         return False
 
     def test_add_new_cluster(self, cluster):
+        '''
+        Add new cluster
+
+        Args:
+            cluster (str): cluster
```
Comment (alvarocarvajald) [inline e2e_test/hawk_test_driver.py:457]: Suggestion:
```suggestion
            cluster (str): cluster name
```

File: e2e_test/hawk_test_results.py [hunk 11]
```diff
@@ -10,27 +10,43 @@
 
 
 class ResultSet:
+    '''
+    Class for handling results data in Hawk GUI test
+    '''
     def __init__(self):
+        '''
+        Constructor function to initlize mandatory data for results_set
```
Comment (alvarocarvajald) [inline e2e_test/hawk_test_results.py:18]: s/initlize/initialize/

File: e2e_test/hawk_test_results.py [hunk 12]
```diff
@@ -10,27 +10,43 @@
...
         self.results_set['info']['results_file'] = 'hawk_test.results'
+        # set initial values for self.results_set['summary']
         self.results_set['summary']['duration'] = 0
         self.results_set['summary']['passed'] = 0
         self.results_set['summary']['num_tests'] = len(self.my_tests)
 
     def add_ssh_tests(self):
+        '''
+        Append callable functions of test cases
+        from HawkTestDriver methods starts with verify_
```
Comment (alvarocarvajald) [inline e2e_test/hawk_test_results.py:48]:
```suggestion
        from HawkTestSSH methods starts with verify_
```

File: e2e_test/hawk_test_driver.py (hunk 1 above)
Comment (Dawei-Pang) [reply e2e_test/hawk_test_driver.py]: Thanks, fixed.

File: e2e_test/hawk_test_driver.py (hunk 2 above)
Comment (Dawei-Pang) [reply e2e_test/hawk_test_driver.py]: Thanks, fixed.

File: e2e_test/hawk_test_driver.py (hunk 3 above)
Comment (Dawei-Pang) [reply e2e_test/hawk_test_driver.py]: Thanks, fixed.

File: e2e_test/hawk_test_driver.py (hunk 4 above)
Comment (Dawei-Pang) [reply e2e_test/hawk_test_driver.py]: Thanks, fixed.

File: e2e_test/hawk_test_driver.py (hunk 5 above)
Comment (Dawei-Pang) [reply e2e_test/hawk_test_driver.py]: Thanks, fixed.

File: e2e_test/hawk_test_driver.py (hunk 6 above)
Comment (Dawei-Pang) [reply e2e_test/hawk_test_driver.py]: Thanks, fixed.

File: e2e_test/hawk_test_driver.py (hunk 7 above)
Comment (Dawei-Pang) [reply e2e_test/hawk_test_driver.py]: Thanks, fixed.

File: e2e_test/hawk_test_driver.py (hunk 8 above)
Comment (Dawei-Pang) [reply e2e_test/hawk_test_driver.py]: Thanks, fixed.

File: e2e_test/hawk_test_driver.py (hunk 9 above)
Comment (Dawei-Pang) [reply e2e_test/hawk_test_driver.py]: Thanks, fixed.

File: e2e_test/hawk_test_driver.py (hunk 10 above)
Comment (Dawei-Pang) [reply e2e_test/hawk_test_driver.py]: Thanks, fixed.

File: e2e_test/hawk_test_results.py (hunk 11 above)
Comment (Dawei-Pang) [reply e2e_test/hawk_test_results.py]: Thanks, fixed.

File: e2e_test/hawk_test_results.py (hunk 12 above)
Comment (Dawei-Pang) [reply e2e_test/hawk_test_results.py]: Thanks, fixed.

File: e2e_test/hawk_test_driver.py [hunk 13]
```diff
@@ -172,6 +224,18 @@ def click_on(self, text):
...
+
+        Args:
+            bywhat (str): 'By.' elements
+            texto  (str):  element name
+            tout   (int):  timeout number, default is 60
+        Returns:
+            boolean: False for failure or element if found
+        Raises:
+            TimeoutException
+        '''
```
[the remark this answers is not in the document]
Comment (Dawei-Pang) [reply e2e_test/hawk_test_driver.py:229]: @aleksei-burlakov Thanks for your suggestion. I have aligned the layout of all comments, would you please help to review again? Thanks!

**Comment by aleksei-burlakov** [comment]
Thanks!

**Pull request merged**